src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -377,12 +386,12 @@ def can_loop_be_parallelised(self, loop, loop_variable=None,
             var_info = var_accesses[signature]
```

Comment:
What I meant was: we could (in theory - you might know that it's not possible, I am not 100% certain) have a copy of a symbol somewhere. So same symbol, but different instances. This would mean that a different hash value is returned, meaning two signatures with the same information, but based on different instances of the symbol, would not be identical anymore. Using strings avoids this issue :)  Otherwise, we would have to add our own hash function to Symbol to be certain.
 As I've said ... maybe you are certain that there is only ever one instance of a symbol, then my argument is moot :) But you also agreed that it certainly the way to go for now. So ... why am I writing that much??